### PR TITLE
[FARM-381] Farmsmart Chat JSON flow

### PR DESCRIPTION
**Pull request by albertmiro**
[FARM-381] 

#### 📲 What

Adding Farmsmart JSON Flow.

#### 🤔 Why
		
To run the correct chatbot Flow for Farmsmart app.
		
#### 🛠 How
		
More in-depth discussion of the change or implementation.

#### 👀 See
		
Screenshots / external resources
		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+              "title": "I don't know"
+            }
+          ]
+        }
+      }
+    },
+    {
+      "text": "When do you intend to start planting?",
+      "inputRequest": {
+        "type": "com.wearemobilefirst.String",
```
Comment (albertmiro) [inline assets/responses/farmsmart_chat_ui_flow.json]: That should be a Date type

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+              "responseText": "I have experience farming"
+            }
+          ]
+        }
+      }
+    },
+    {
+      "text": "Where is your farm?",
+      "inputRequest": {
+        "type": "com.wearemobilefirst.String",
```
Comment (albertmiro) [inline assets/responses/farmsmart_chat_ui_flow.json]: That should be a Dropdown type

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+      "sentiment": "Positive"
+    },
+    {
+      "text": "What is your name?",
+      "inputRequest": {
+        "type": "com.wearemobilefirst.String",
+        "uri": "%Name",
+        "localStore": false,
+        "inline": true,
+        "title": "Your Name",
```
Comment (albertmiro) [inline assets/responses/farmsmart_chat_ui_flow.json:38]: Not sure how we will translate titles or text that comes from JSON with `Intl.message` 🤔 We will have different JSON depending on language shown?

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+    {
+      "text": "Where is your farm?",
+      "inputRequest": {
+        "type": "com.wearemobilefirst.Dropdown",
+        "uri": "%FarmLocation",
+        "title": "Farm Location",
+        "localStore": false,
+        "responseText": "",
+        "inline": true,
+        "validationRegex": "^[a-zA-Z]+$",
```
Comment (alexanishchenko88) [inline assets/responses/farmsmart_chat_ui_flow.json]: Please remove this regexp as in this case we will NOT use it.

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+    },
+    {
+      "text": "When do you intend to start planting?",
+      "inputRequest": {
+        "type": "com.wearemobilefirst.Date",
+        "uri": "%StartDate",
+        "localStore": false,
+        "inline": true,
+        "title": "Start Date",
+        "validationRegex": "((0[1-9]|[12]\\d|3[01])/(0[1-9]|1[0-2])/[12]\\d{3})",
```
Comment (alexanishchenko88) [inline assets/responses/farmsmart_chat_ui_flow.json]: The same here for regexp

**Review by alexanishchenko88 — CHANGES_REQUESTED** [review]

File: assets/responses/farmsmart_chat_ui_flow.json (hunk 4 above)
Comment (albertmiro) [reply assets/responses/farmsmart_chat_ui_flow.json]: Done

File: assets/responses/farmsmart_chat_ui_flow.json (hunk 5 above)
Comment (albertmiro) [reply assets/responses/farmsmart_chat_ui_flow.json]: Done

**Review by alexanishchenko88 — DISMISSED** [review]

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+    {
+      "text": "Where is your farm?",
+      "inputRequest": {
+        "type": "com.wearemobilefirst.Dropdown",
+        "uri": "%FarmLocation",
+        "title": "Farm Location",
+        "localStore": false,
+        "responseText": "",
+        "inline": true,
+        "optional": false
```
Comment (alexanishchenko88) [inline assets/responses/farmsmart_chat_ui_flow.json]: I think we should have arg here with option to pick when drop down appears. The structure should be as in MultiChoice one i guess...

Comment (albertmiro) [reply assets/responses/farmsmart_chat_ui_flow.json]: True! Thanks

Comment (albertmiro) [reply assets/responses/farmsmart_chat_ui_flow.json]: Done

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 7]
```diff
@@ -0,0 +1,535 @@
...
+            {
+              "id": "OptionKey45",
+              "title": "Vihiga"
+            },
+            {
+              "id": "OptionKey46",
+              "title": "Wajir"
+            },
+            {
+              "id": "OptionKey4",
```
Comment (alexanishchenko88) [inline assets/responses/farmsmart_chat_ui_flow.json]:
```suggestion
              "id": "OptionKey47",
```

**Review by alexanishchenko88 — DISMISSED** [review]

**Review by davidfranquet — DISMISSED** [review]

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 8]
```diff
@@ -0,0 +1,535 @@
+{
+  "uid": "1",
+  "title": "Jata",
+  "subtitle": "Your personal farm assistant",
+  "origin": "com.wearemobilefirst.farmsmart",
+  "formResponse": {
+    "type": "com.google.firebase.database",
```
Comment (LeeMatthewHiggins) [inline assets/responses/farmsmart_chat_ui_flow.json]: com.wearemobilefirst.local

File: assets/responses/farmsmart_chat_ui_flow.json [hunk 9]
```diff
@@ -0,0 +1,535 @@
...
+      "name": "%Name",
+      "skillLevel": "%SkillLevel",
+      "farmLocation": "%FarmLocation",
+      "agroEcoZone": "%AgroEcoZone",
+      "landSize": "%LandSize",
+      "motivation": "%Motivation",
+      "peopleInHousehold": "%PeopleInHousehold",
+      "soilType": "%SoilType",
+      "startDate": "%StartDate",
+      "usingIrrigation": "%UsingIrrigation",
```
Comment (alexanishchenko88) [inline assets/responses/farmsmart_chat_ui_flow.json]:
```suggestion
      "usingIrrigation": "%UsingIrrigation"
```

File: assets/responses/farmsmart_chat_ui_flow.json (hunk 3 above)
Comment (LeeMatthewHiggins) [reply assets/responses/farmsmart_chat_ui_flow.json:38]: yes we'll have a whole file per language (as this would normally be server side driven).

**Review by LeeMatthewHiggins — APPROVED** [review]
LGTM 👍

**Review by alexanishchenko88 — APPROVED** [review]

**Pull request merged**